src/mappers/unit/unit_mapper.cpp

```diff
@@ -0,0 +1,41 @@
+#include "src/mappers/unit/unit_mapper.h"
+
+#include <algorithm>
+#include <ranges>
+
+#include "external/fmt/include/fmt/format.h"
+
+std::vector<hoi4::Battalion> mappers::UnitMapper::MakeBattalions(const std::vector<std::string> methods,
+    int scale) const
+{
+   static std::set<std::string> warned;
+
+   int equip = 0;
+   BattalionMap current;
+   for (const auto& pm: methods)
+   {
+      const auto& itr = templates_.find(pm);
+      if (itr == templates_.end())
+      {
+         if (warned.find(pm) == warned.end())
+         {
+            warned.insert(pm);
+            Log(LogLevel::Warning) << fmt::format("Missing unit mapping rule for {}", pm);
+         }
+         continue;
+      }
```

Comment:
Warning should be moved to a dedicated function so it doesn't distract from this one.